### PR TITLE
[#8166] Fix ClassNotFoundException on jdk6

### LINKED ISSUE #8166: In netty-4.1.28.Final version, an error occurs when running on jdk6.
### Actual behavior
```
io.netty.channel.ChannelException: Unable to create Channel from class class io.netty.channel.socket.nio.NioSocketChannel
Caused by: java.lang.NoClassDefFoundError: java/nio/channels/NetworkChannel
	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:105)
	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:94)
	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:87)
	at io.netty.channel.socket.nio.NioSocketChannel.<init>(NioSocketChannel.java:80)
	... 32 more
Caused by: java.lang.ClassNotFoundException: java.nio.channels.NetworkChannel
	at java.net.URLClassLoader$1.run(URLClassLoader.java:202)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
```

### Steps to reproduce
```
// run following test on jvm6

    @Test
    public void listenerTest() throws Exception {
        Bootstrap bootstrap = createBootstrap();
        Channel channel = bootstrap.connect(host, port).sync().channel();

    }

    public Bootstrap createBootstrap() {
        EventLoopGroup workerGroup = new NioEventLoopGroup();
        Bootstrap bootstrap = new Bootstrap();
        bootstrap.group(workerGroup).channel(NioSocketChannel.class)
                .handler(new ChannelInitializer<SocketChannel>() {
                    @Override
                    protected void initChannel(SocketChannel ch) throws Exception {
                        ch.pipeline().addLast(new HttpClientCodec());
                        ch.pipeline().addLast(new HttpObjectAggregator(65535));
                    }
                });
        return bootstrap;
    }
```

### Netty version
netty-4.1.28.Final

### JVM version (e.g. `java -version`)
1.6.0_45

### OS version (e.g. `uname -a`)
Windows10,  CentOS 6.6

I think this issue is problem, but I'm not sure.
If this issue is problem, May I make a correction?

### PR DESCRIPTION
**Pull request by koo-taejin**
Motivation:
Fixed problem loading class included in java7+ when using java6

Modification:
Using interface, java6 and java7 behave differently.

Result:
Fixes #8166

### CODE REVIEW

File: transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java [hunk 1]
```diff
@@ -525,4 +534,87 @@ private SocketChannel jdkChannel() {
...
+    private final class NioChannelOptionHelper implements SocketChannelOptionHelper {
+
+        @Override
+        public boolean isSupported() {
+            return PlatformDependent.javaVersion() >= 7;
+        }
+
+        @Override
+        public boolean isSupported(ChannelOption option) {
+            if (isSupported() && option instanceof NioChannelOption) {
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java]: Just use

return isSupported() && option instanceof NioChannelOption

Comment (koo-taejin) [reply transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java]: I have corrected what you said. :)

**Comment by koo-taejin** [comment]
@normanmaurer I'm glad you found an easier way.

**Comment by normanmaurer** [comment]
Let me close this... 
https://github.com/netty/netty/pull/8168 should fix it...

**Pull request closed without merging**